libdrizzle/result.cc

```diff
@@ -464,19 +464,11 @@ drizzle_return_t drizzle_state_result_read(drizzle_st *con)
     con->packet_size-= 5;
     ret= DRIZZLE_RETURN_OK;
   }
-  else if (con->buffer_ptr[0] == 255)
+  else if (drizzle_check_unpack_error(con))
   {
-    con->result->error_code= drizzle_get_byte2(con->buffer_ptr + 1);
-    con->error_code= con->result->error_code;
-    /* Byte 3 is always a '#' character, skip it. */
-    memcpy(con->result->sqlstate, con->buffer_ptr + 4,
+    memcpy(con->result->sqlstate, con->sqlstate,
```

Comment:
Why are we changing this `memcpy`? Previously `con->sqlstate` was copied to `con->sqlstate`.